In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
crime=pd.read_csv("crimes_against_women_2001-2014.csv")

In [3]:
crime

Unnamed: 0        STATE/UT           DISTRICT  Year  Rape  \
0               0  ANDHRA PRADESH           ADILABAD  2001    50   
1               1  ANDHRA PRADESH          ANANTAPUR  2001    23   
2               2  ANDHRA PRADESH           CHITTOOR  2001    27   
3               3  ANDHRA PRADESH           CUDDAPAH  2001    20   
4               4  ANDHRA PRADESH      EAST GODAVARI  2001    23   
...           ...             ...                ...   ...   ...   
10672         832     Lakshadweep        Lakshadweep  2014     1   
10673         833     Lakshadweep  Total District(s)  2014     1   
10674         834      Puducherry           Karaikal  2014     3   
10675         835      Puducherry         Puducherry  2014     7   
10676         836      Puducherry  Total District(s)  2014    10   

       Kidnapping and Abduction  Dowry Deaths  \
0                            30            16   
1                            30             7   
2                            34            14   
3                            20            17   
4                            26            12   
...                         ...           ...   
10672                         0             0   
10673                         0             0   
10674                         1             0   
10675                         6             1   
10676                         7             1   

       Assault on women with intent to outrage her modesty  \
0                                                    149     
1                                                    118     
2                                                    112     
3                                                    126     
4                                                    109     
...                                                  ...     
10672                                                  1     
10673                                                  1     
10674                                                 12     
10675                                                 20     
10676                                                 32     

       Insult to modesty of Women  Cruelty by Husband or his Relatives  \
0                              34                                  175   
1                              24                                  154   
2                              83                                  186   
3                              38                                   57   
4                              58                                  247   
...                           ...                                  ...   
10672                           2                                    0   
10673                           2                                    0   
10674                           1                                    1   
10675                           7                                    3   
10676                           8                                    4   

       Importation of Girls  
0                         0  
1                         0  
2                         0  
3                         0  
4                         0  
...                     ...  
10672                     0  
10673                     0  
10674                     0  
10675                     0  
10676                     0  

[10677 rows x 11 columns]

In [4]:
crime.shape

(10677, 11)

In [5]:
crime.columns

Index(['Unnamed: 0', 'STATE/UT', 'DISTRICT', 'Year', 'Rape',
       'Kidnapping and Abduction', 'Dowry Deaths',
       'Assault on women with intent to outrage her modesty',
       'Insult to modesty of Women', 'Cruelty by Husband or his Relatives',
       'Importation of Girls'],
      dtype='object')

In [6]:
crime.head()

Unnamed: 0        STATE/UT       DISTRICT  Year  Rape  \
0           0  ANDHRA PRADESH       ADILABAD  2001    50   
1           1  ANDHRA PRADESH      ANANTAPUR  2001    23   
2           2  ANDHRA PRADESH       CHITTOOR  2001    27   
3           3  ANDHRA PRADESH       CUDDAPAH  2001    20   
4           4  ANDHRA PRADESH  EAST GODAVARI  2001    23   

   Kidnapping and Abduction  Dowry Deaths  \
0                        30            16   
1                        30             7   
2                        34            14   
3                        20            17   
4                        26            12   

   Assault on women with intent to outrage her modesty  \
0                                                149     
1                                                118     
2                                                112     
3                                                126     
4                                                109     

   Insult to modesty of Women  Cruelty by Husband or his Relatives  \
0                          34                                  175   
1                          24                                  154   
2                          83                                  186   
3                          38                                   57   
4                          58                                  247   

   Importation of Girls  
0                     0  
1                     0  
2                     0  
3                     0  
4                     0

In [7]:
crime.tail()

Unnamed: 0     STATE/UT           DISTRICT  Year  Rape  \
10672         832  Lakshadweep        Lakshadweep  2014     1   
10673         833  Lakshadweep  Total District(s)  2014     1   
10674         834   Puducherry           Karaikal  2014     3   
10675         835   Puducherry         Puducherry  2014     7   
10676         836   Puducherry  Total District(s)  2014    10   

       Kidnapping and Abduction  Dowry Deaths  \
10672                         0             0   
10673                         0             0   
10674                         1             0   
10675                         6             1   
10676                         7             1   

       Assault on women with intent to outrage her modesty  \
10672                                                  1     
10673                                                  1     
10674                                                 12     
10675                                                 20     
10676                                                 32     

       Insult to modesty of Women  Cruelty by Husband or his Relatives  \
10672                           2                                    0   
10673                           2                                    0   
10674                           1                                    1   
10675                           7                                    3   
10676                           8                                    4   

       Importation of Girls  
10672                     0  
10673                     0  
10674                     0  
10675                     0  
10676                     0

In [8]:
crime['STATE/UT'].nunique()

72

In [9]:
crime['DISTRICT'].nunique()

1605

In [10]:
crime['Year'].nunique()

14

In [11]:
crime['Rape'].nunique()

481

In [12]:
crime['Kidnapping and Abduction'].nunique()

555

In [13]:
crime.isnull().sum()

Unnamed: 0                                             0
STATE/UT                                               0
DISTRICT                                               0
Year                                                   0
Rape                                                   0
Kidnapping and Abduction                               0
Dowry Deaths                                           0
Assault on women with intent to outrage her modesty    0
Insult to modesty of Women                             0
Cruelty by Husband or his Relatives                    0
Importation of Girls                                   0
dtype: int64

In [14]:
crime.drop('Unnamed: 0',axis=1,inplace=True)

In [15]:
crime

STATE/UT           DISTRICT  Year  Rape  \
0      ANDHRA PRADESH           ADILABAD  2001    50   
1      ANDHRA PRADESH          ANANTAPUR  2001    23   
2      ANDHRA PRADESH           CHITTOOR  2001    27   
3      ANDHRA PRADESH           CUDDAPAH  2001    20   
4      ANDHRA PRADESH      EAST GODAVARI  2001    23   
...               ...                ...   ...   ...   
10672     Lakshadweep        Lakshadweep  2014     1   
10673     Lakshadweep  Total District(s)  2014     1   
10674      Puducherry           Karaikal  2014     3   
10675      Puducherry         Puducherry  2014     7   
10676      Puducherry  Total District(s)  2014    10   

       Kidnapping and Abduction  Dowry Deaths  \
0                            30            16   
1                            30             7   
2                            34            14   
3                            20            17   
4                            26            12   
...                         ...           ...   
10672                         0             0   
10673                         0             0   
10674                         1             0   
10675                         6             1   
10676                         7             1   

       Assault on women with intent to outrage her modesty  \
0                                                    149     
1                                                    118     
2                                                    112     
3                                                    126     
4                                                    109     
...                                                  ...     
10672                                                  1     
10673                                                  1     
10674                                                 12     
10675                                                 20     
10676                                                 32     

       Insult to modesty of Women  Cruelty by Husband or his Relatives  \
0                              34                                  175   
1                              24                                  154   
2                              83                                  186   
3                              38                                   57   
4                              58                                  247   
...                           ...                                  ...   
10672                           2                                    0   
10673                           2                                    0   
10674                           1                                    1   
10675                           7                                    3   
10676                           8                                    4   

       Importation of Girls  
0                         0  
1                         0  
2                         0  
3                         0  
4                         0  
...                     ...  
10672                     0  
10673                     0  
10674                     0  
10675                     0  
10676                     0  

[10677 rows x 10 columns]

In [16]:
crime['DISTRICT'].nunique()

1605

In [17]:
districts=len(crime.DISTRICT.unique())

In [18]:
districts

1605

There are 718 districts in India, but in this data there are 1605 districts.

In [19]:
crime.drop('DISTRICT',axis=1,inplace=True)

In [20]:
crime

STATE/UT  Year  Rape  Kidnapping and Abduction  Dowry Deaths  \
0      ANDHRA PRADESH  2001    50                        30            16   
1      ANDHRA PRADESH  2001    23                        30             7   
2      ANDHRA PRADESH  2001    27                        34            14   
3      ANDHRA PRADESH  2001    20                        20            17   
4      ANDHRA PRADESH  2001    23                        26            12   
...               ...   ...   ...                       ...           ...   
10672     Lakshadweep  2014     1                         0             0   
10673     Lakshadweep  2014     1                         0             0   
10674      Puducherry  2014     3                         1             0   
10675      Puducherry  2014     7                         6             1   
10676      Puducherry  2014    10                         7             1   

       Assault on women with intent to outrage her modesty  \
0                                                    149     
1                                                    118     
2                                                    112     
3                                                    126     
4                                                    109     
...                                                  ...     
10672                                                  1     
10673                                                  1     
10674                                                 12     
10675                                                 20     
10676                                                 32     

       Insult to modesty of Women  Cruelty by Husband or his Relatives  \
0                              34                                  175   
1                              24                                  154   
2                              83                                  186   
3                              38                                   57   
4                              58                                  247   
...                           ...                                  ...   
10672                           2                                    0   
10673                           2                                    0   
10674                           1                                    1   
10675                           7                                    3   
10676                           8                                    4   

       Importation of Girls  
0                         0  
1                         0  
2                         0  
3                         0  
4                         0  
...                     ...  
10672                     0  
10673                     0  
10674                     0  
10675                     0  
10676                     0  

[10677 rows x 9 columns]

In [21]:
crime.rename(columns={'Kidnapping and Abduction':'Kidnapping_Abduction', 'Dowry Deaths':'Dowry_Deaths',
       'Assault on women with intent to outrage her modesty':'Assault_on_women_with_intent_to_outrage_her_modesty',
       'Insult to modesty of Women':'Insult_to_modesty_of_Women', 'Cruelty by Husband or his Relatives':'Cruelty_by_Husband_or_his_Relatives',
       'Importation of Girls':'Importation_of_Girls'},inplace=True)

In [22]:
crime

STATE/UT  Year  Rape  Kidnapping_Abduction  Dowry_Deaths  \
0      ANDHRA PRADESH  2001    50                    30            16   
1      ANDHRA PRADESH  2001    23                    30             7   
2      ANDHRA PRADESH  2001    27                    34            14   
3      ANDHRA PRADESH  2001    20                    20            17   
4      ANDHRA PRADESH  2001    23                    26            12   
...               ...   ...   ...                   ...           ...   
10672     Lakshadweep  2014     1                     0             0   
10673     Lakshadweep  2014     1                     0             0   
10674      Puducherry  2014     3                     1             0   
10675      Puducherry  2014     7                     6             1   
10676      Puducherry  2014    10                     7             1   

       Assault_on_women_with_intent_to_outrage_her_modesty  \
0                                                    149     
1                                                    118     
2                                                    112     
3                                                    126     
4                                                    109     
...                                                  ...     
10672                                                  1     
10673                                                  1     
10674                                                 12     
10675                                                 20     
10676                                                 32     

       Insult_to_modesty_of_Women  Cruelty_by_Husband_or_his_Relatives  \
0                              34                                  175   
1                              24                                  154   
2                              83                                  186   
3                              38                                   57   
4                              58                                  247   
...                           ...                                  ...   
10672                           2                                    0   
10673                           2                                    0   
10674                           1                                    1   
10675                           7                                    3   
10676                           8                                    4   

       Importation_of_Girls  
0                         0  
1                         0  
2                         0  
3                         0  
4                         0  
...                     ...  
10672                     0  
10673                     0  
10674                     0  
10675                     0  
10676                     0  

[10677 rows x 9 columns]

In [23]:
print(crime['STATE/UT'].nunique())

72


In [24]:
print(crime['STATE/UT'].unique())

['ANDHRA PRADESH' 'ARUNACHAL PRADESH' 'ASSAM' 'BIHAR' 'CHHATTISGARH' 'GOA'
 'GUJARAT' 'HARYANA' 'HIMACHAL PRADESH' 'JAMMU & KASHMIR' 'JHARKHAND'
 'KARNATAKA' 'KERALA' 'MADHYA PRADESH' 'MAHARASHTRA' 'MANIPUR' 'MEGHALAYA'
 'MIZORAM' 'NAGALAND' 'ODISHA' 'PUNJAB' 'RAJASTHAN' 'SIKKIM' 'TAMIL NADU'
 'TRIPURA' 'UTTAR PRADESH' 'UTTARAKHAND' 'WEST BENGAL' 'A & N ISLANDS'
 'CHANDIGARH' 'D & N HAVELI' 'DAMAN & DIU' 'DELHI' 'LAKSHADWEEP'
 'PUDUCHERRY' 'Andhra Pradesh' 'Arunachal Pradesh' 'Assam' 'Bihar'
 'Chhattisgarh' 'Goa' 'Gujarat' 'Haryana' 'Himachal Pradesh'
 'Jammu & Kashmir' 'Jharkhand' 'Karnataka' 'Kerala' 'Madhya Pradesh'
 'Maharashtra' 'Manipur' 'Meghalaya' 'Mizoram' 'Nagaland' 'Odisha'
 'Punjab' 'Rajasthan' 'Sikkim' 'Tamil Nadu' 'Tripura' 'Uttar Pradesh'
 'Uttarakhand' 'West Bengal' 'A&N Islands' 'Chandigarh' 'D&N Haveli'
 'Daman & Diu' 'Delhi UT' 'Lakshadweep' 'Puducherry' 'Telangana'
 'A & N Islands']


In [25]:
#first we will remove all the repeated words.

In [26]:
def remove_uppercase(r):
    r=r['STATE/UT'].strip()
    r=r.upper()
    return r
crime['STATE/UT']=crime.apply(remove_uppercase,axis=1)

In [27]:
crime['STATE/UT'].unique()

array(['ANDHRA PRADESH', 'ARUNACHAL PRADESH', 'ASSAM', 'BIHAR',
       'CHHATTISGARH', 'GOA', 'GUJARAT', 'HARYANA', 'HIMACHAL PRADESH',
       'JAMMU & KASHMIR', 'JHARKHAND', 'KARNATAKA', 'KERALA',
       'MADHYA PRADESH', 'MAHARASHTRA', 'MANIPUR', 'MEGHALAYA', 'MIZORAM',
       'NAGALAND', 'ODISHA', 'PUNJAB', 'RAJASTHAN', 'SIKKIM',
       'TAMIL NADU', 'TRIPURA', 'UTTAR PRADESH', 'UTTARAKHAND',
       'WEST BENGAL', 'A & N ISLANDS', 'CHANDIGARH', 'D & N HAVELI',
       'DAMAN & DIU', 'DELHI', 'LAKSHADWEEP', 'PUDUCHERRY', 'A&N ISLANDS',
       'D&N HAVELI', 'DELHI UT', 'TELANGANA'], dtype=object)

In [28]:
crime['STATE/UT'].nunique()

39

In [29]:
crime

STATE/UT  Year  Rape  Kidnapping_Abduction  Dowry_Deaths  \
0      ANDHRA PRADESH  2001    50                    30            16   
1      ANDHRA PRADESH  2001    23                    30             7   
2      ANDHRA PRADESH  2001    27                    34            14   
3      ANDHRA PRADESH  2001    20                    20            17   
4      ANDHRA PRADESH  2001    23                    26            12   
...               ...   ...   ...                   ...           ...   
10672     LAKSHADWEEP  2014     1                     0             0   
10673     LAKSHADWEEP  2014     1                     0             0   
10674      PUDUCHERRY  2014     3                     1             0   
10675      PUDUCHERRY  2014     7                     6             1   
10676      PUDUCHERRY  2014    10                     7             1   

       Assault_on_women_with_intent_to_outrage_her_modesty  \
0                                                    149     
1                                                    118     
2                                                    112     
3                                                    126     
4                                                    109     
...                                                  ...     
10672                                                  1     
10673                                                  1     
10674                                                 12     
10675                                                 20     
10676                                                 32     

       Insult_to_modesty_of_Women  Cruelty_by_Husband_or_his_Relatives  \
0                              34                                  175   
1                              24                                  154   
2                              83                                  186   
3                              38                                   57   
4                              58                                  247   
...                           ...                                  ...   
10672                           2                                    0   
10673                           2                                    0   
10674                           1                                    1   
10675                           7                                    3   
10676                           8                                    4   

       Importation_of_Girls  
0                         0  
1                         0  
2                         0  
3                         0  
4                         0  
...                     ...  
10672                     0  
10673                     0  
10674                     0  
10675                     0  
10676                     0  

[10677 rows x 9 columns]

In [30]:
#now use replace function to replace the other type of repeated datas as discussed above

crime['STATE/UT'].replace("A&N ISLANDS","A & N ISLANDS",inplace=True)
crime['STATE/UT'].replace("D&N HAVELI","D & N HAVELI",inplace=True)
crime['STATE/UT'].replace("DELHI UT","DELHI",inplace=True)

In [31]:
crime['STATE/UT'].nunique()

36

In [32]:
crime

STATE/UT  Year  Rape  Kidnapping_Abduction  Dowry_Deaths  \
0      ANDHRA PRADESH  2001    50                    30            16   
1      ANDHRA PRADESH  2001    23                    30             7   
2      ANDHRA PRADESH  2001    27                    34            14   
3      ANDHRA PRADESH  2001    20                    20            17   
4      ANDHRA PRADESH  2001    23                    26            12   
...               ...   ...   ...                   ...           ...   
10672     LAKSHADWEEP  2014     1                     0             0   
10673     LAKSHADWEEP  2014     1                     0             0   
10674      PUDUCHERRY  2014     3                     1             0   
10675      PUDUCHERRY  2014     7                     6             1   
10676      PUDUCHERRY  2014    10                     7             1   

       Assault_on_women_with_intent_to_outrage_her_modesty  \
0                                                    149     
1                                                    118     
2                                                    112     
3                                                    126     
4                                                    109     
...                                                  ...     
10672                                                  1     
10673                                                  1     
10674                                                 12     
10675                                                 20     
10676                                                 32     

       Insult_to_modesty_of_Women  Cruelty_by_Husband_or_his_Relatives  \
0                              34                                  175   
1                              24                                  154   
2                              83                                  186   
3                              38                                   57   
4                              58                                  247   
...                           ...                                  ...   
10672                           2                                    0   
10673                           2                                    0   
10674                           1                                    1   
10675                           7                                    3   
10676                           8                                    4   

       Importation_of_Girls  
0                         0  
1                         0  
2                         0  
3                         0  
4                         0  
...                     ...  
10672                     0  
10673                     0  
10674                     0  
10675                     0  
10676                     0  

[10677 rows x 9 columns]

In [33]:
crime.columns

Index(['STATE/UT', 'Year', 'Rape', 'Kidnapping_Abduction', 'Dowry_Deaths',
       'Assault_on_women_with_intent_to_outrage_her_modesty',
       'Insult_to_modesty_of_Women', 'Cruelty_by_Husband_or_his_Relatives',
       'Importation_of_Girls'],
      dtype='object')

In [34]:
victims_raped=crime.Rape.sum()

In [35]:
victims_raped

619158

In [36]:
victims_kidnapped=crime.Kidnapping_Abduction.sum()

In [37]:
victims_kidnapped

746198

In [38]:
victims_dowry_deaths=crime.Dowry_Deaths.sum()

In [39]:
victims_dowry_deaths

215480

In [40]:
victims_assault=crime.Assault_on_women_with_intent_to_outrage_her_modesty.sum()

In [41]:
victims_assault

1212258

In [42]:
victims_insult=crime.Insult_to_modesty_of_Women.sum()

In [43]:
victims_insult

292756

In [44]:
victims_cruelty=crime.Cruelty_by_Husband_or_his_Relatives.sum()

In [45]:
victims_cruelty

2233888

In [46]:
victims_importation=crime.Importation_of_Girls.sum()

In [47]:
victims_importation

1872

In [48]:
total_population=victims_raped+victims_kidnapped+victims_dowry_deaths+victims_assault+victims_insult+victims_cruelty+victims_importation

#### TOTAL POPULATION

In [49]:
total_population

5321610

In [50]:
crime.sort_values(by='Rape',ascending=False).head(5) # FIND TOP 5 STATE WHERE MAXIMUM RAPE 

STATE/UT  Year  Rape  Kidnapping_Abduction  Dowry_Deaths  \
10244  MADHYA PRADESH  2014  5076                  5688           733   
9426   MADHYA PRADESH  2013  4335                  2873           776   
10445       RAJASTHAN  2014  3759                  4421           408   
10595   UTTAR PRADESH  2014  3467                 10626          2469   
10291     MAHARASHTRA  2014  3438                  2457           279   

       Assault_on_women_with_intent_to_outrage_her_modesty  \
10244                                               9609     
9426                                                8252     
10445                                               5999     
10595                                               8605     
10291                                              10001     

       Insult_to_modesty_of_Women  Cruelty_by_Husband_or_his_Relatives  \
10244                         429                                 6451   
9426                          736                                 4988   
10445                          18                                15905   
10595                          29                                10471   
10291                        1575                                 7696   

       Importation_of_Girls  
10244                     0  
9426                      7  
10445                     0  
10595                     0  
10291                     1

In [51]:
crime.sort_values(by='Importation_of_Girls',ascending=False).head(5) # FIND TOP 5 STATE WHERE MAXIMUM 'Importation_of_Girls' 

STATE/UT  Year  Rape  Kidnapping_Abduction  Dowry_Deaths  \
115         BIHAR  2001   888                   518           859   
3013        BIHAR  2005  1147                   929          1014   
3597  WEST BENGAL  2005  1686                  1039           446   
3590  WEST BENGAL  2005   148                    97            48   
4486        BIHAR  2007  1555                  1260          1172   

      Assault_on_women_with_intent_to_outrage_her_modesty  \
115                                                 562     
3013                                                451     
3597                                               1572     
3590                                                 95     
4486                                                853     

      Insult_to_modesty_of_Women  Cruelty_by_Husband_or_his_Relatives  \
115                           21                                 1558   
3013                          13                                 1574   
3597                          54                                 6936   
3590                           3                                  545   
4486                          12                                 1635   

      Importation_of_Girls  
115                     83  
3013                    74  
3597                    61  
3590                    60  
4486                    56

# *Questions and Solutions with Observation* 

### Q 1.Create a dataframe containing 10 highest reported rape cases in India, in the span of year 2001-2014.


In [52]:
highest_rape=crime.sort_values('Rape',ascending=False).head(10)

In [53]:
highest_rape

STATE/UT  Year  Rape  Kidnapping_Abduction  Dowry_Deaths  \
10244  MADHYA PRADESH  2014  5076                  5688           733   
9426   MADHYA PRADESH  2013  4335                  2873           776   
10445       RAJASTHAN  2014  3759                  4421           408   
10595   UTTAR PRADESH  2014  3467                 10626          2469   
10291     MAHARASHTRA  2014  3438                  2457           279   
8611   MADHYA PRADESH  2012  3425                  1127           743   
7810   MADHYA PRADESH  2011  3406                  1088           811   
9628        RAJASTHAN  2013  3285                  4047           453   
7025   MADHYA PRADESH  2010  3135                  1030           892   
9472      MAHARASHTRA  2013  3063                  1874           320   

       Assault_on_women_with_intent_to_outrage_her_modesty  \
10244                                               9609     
9426                                                8252     
10445                                               5999     
10595                                               8605     
10291                                              10001     
8611                                                6655     
7810                                                6665     
9628                                                4829     
7025                                                6646     
9472                                                8132     

       Insult_to_modesty_of_Women  Cruelty_by_Husband_or_his_Relatives  \
10244                         429                                 6451   
9426                          736                                 4988   
10445                          18                                15905   
10595                          29                                10471   
10291                        1575                                 7696   
8611                          774                                 3988   
7810                          762                                 3732   
9628                           25                                15094   
7025                          918                                 3756   
9472                         2632                                 8542   

       Importation_of_Girls  
10244                     0  
9426                      7  
10445                     0  
10595                     0  
10291                     1  
8611                      6  
7810                     45  
9628                      1  
7025                      5  
9472                      0

#### *After Analysis this data the top states that reported the maximum number of rape cases*
#### *Where Madhya Pradesh reported the maximum rape cases in the year 2014.*

### Q2.Create a dataframe containing 10 highest reported deaths caused by Dowry cases in India, in the span of year 2001-2014.

In [54]:
highest_death=crime.sort_values('Dowry_Deaths',ascending=False).head(10)

In [55]:
highest_death

STATE/UT  Year  Rape  Kidnapping_Abduction  Dowry_Deaths  \
10595  UTTAR PRADESH  2014  3467                 10626          2469   
9760   UTTAR PRADESH  2013  3050                  9737          2335   
8132   UTTAR PRADESH  2011  2042                  7525          2322   
8938   UTTAR PRADESH  2012  1963                  7910          2244   
5796   UTTAR PRADESH  2008  1871                  4439          2237   
6563   UTTAR PRADESH  2009  1759                  5078          2232   
7342   UTTAR PRADESH  2010  1563                  5468          2217   
650    UTTAR PRADESH  2001  1958                  2879          2211   
5040   UTTAR PRADESH  2007  1648                  3363          2076   
1366   UTTAR PRADESH  2002  1415                  2298          1893   

       Assault_on_women_with_intent_to_outrage_her_modesty  \
10595                                               8605     
9760                                                7303     
8132                                                3455     
8938                                                3247     
5796                                                2955     
6563                                                2782     
7342                                                2793     
650                                                 2870     
5040                                                2522     
1366                                                2145     

       Insult_to_modesty_of_Women  Cruelty_by_Husband_or_his_Relatives  \
10595                          29                                10471   
9760                           25                                 8781   
8132                            3                                 7121   
8938                            8                                 7661   
5796                         3374                                 8312   
6563                         2524                                 8566   
7342                           11                                 7978   
650                          2575                                 7365   
5040                         2882                                 7650   
1366                         1887                                 5679   

       Importation_of_Girls  
10595                     0  
9760                      0  
8132                      0  
8938                      0  
5796                      0  
6563                      0  
7342                      0  
650                       0  
5040                      0  
1366                      0

#### According to Analysis this data the heighest 'deaths caused by Dowry'  was UTTAR PRADESH in 2014 with number of cases reported being 2469.


## Q3.Create a dataframe containing 10 highest reported Domestic Violance cases in India, in the span of year 2001-2014.

In [56]:
highest_Domestic_v=crime.sort.values('Domestic_Violance',ascending=False).head(10)

AttributeError: 'DataFrame' object has no attribute 'sort'

## Q4. Create a dataframe containing 10 highest reported Importation cases in India, in the span of year 2001-2014.

In [57]:
highest_importation=crime.sort_values('Importation_of_Girls' ,ascending=False).head(10)

In [58]:
highest_importation

STATE/UT  Year  Rape  Kidnapping_Abduction  Dowry_Deaths  \
115            BIHAR  2001   888                   518           859   
3013           BIHAR  2005  1147                   929          1014   
3597     WEST BENGAL  2005  1686                  1039           446   
3590     WEST BENGAL  2005   148                    97            48   
4486           BIHAR  2007  1555                  1260          1172   
3005           BIHAR  2005    28                     4            40   
7810  MADHYA PRADESH  2011  3406                  1088           811   
3746           BIHAR  2006  1232                  1084          1188   
102            BIHAR  2001    27                    11            90   
5378       JHARKHAND  2008   791                   499           266   

      Assault_on_women_with_intent_to_outrage_her_modesty  \
115                                                 562     
3013                                                451     
3597                                               1572     
3590                                                 95     
4486                                                853     
3005                                                 10     
7810                                               6665     
3746                                                530     
102                                                  47     
5378                                                271     

      Insult_to_modesty_of_Women  Cruelty_by_Husband_or_his_Relatives  \
115                           21                                 1558   
3013                          13                                 1574   
3597                          54                                 6936   
3590                           3                                  545   
4486                          12                                 1635   
3005                           0                                   73   
7810                         762                                 3732   
3746                          53                                 1689   
102                            0                                  152   
5378                          23                                  851   

      Importation_of_Girls  
115                     83  
3013                    74  
3597                    61  
3590                    60  
4486                    56  
3005                    48  
7810                    45  
3746                    42  
102                     39  
5378                    39

### According to our analysis, maximum of Importation of girls has been reported in Bihar in the year 2011 with number of cases reported being 83.

## Q5.Find out the total number of cases, in span of 2001-2014 under each category, state wise.


In [59]:
total_case_state=crime.groupby('STATE/UT')[['Rape','Kidnapping_Abduction','Dowry_Deaths','Assault_on_women_with_intent_to_outrage_her_modesty','Insult_to_modesty_of_Women',
                                          'Cruelty_by_Husband_or_his_Relatives','Importation_of_Girls' ]].sum()

In [60]:
total_case_state

Rape  Kidnapping_Abduction  Dowry_Deaths  \
STATE/UT                                                       
A & N ISLANDS        336                   212            20   
ANDHRA PRADESH     32150                 34504         13844   
ARUNACHAL PRADESH   1316                  1470             6   
ASSAM              40190                 62074          3268   
BIHAR              30758                 57086         32206   
CHANDIGARH           770                  1682            90   
CHHATTISGARH       29308                 11808          2758   
D & N HAVELI         132                   224             2   
DAMAN & DIU           60                    44             6   
DELHI              20312                 46586          3758   
GOA                 1062                   640            38   
GUJARAT            11644                 34670          1108   
HARYANA            17110                 20016          7372   
HIMACHAL PRADESH    4674                  4116           112   
JAMMU & KASHMIR     7038                 21164           294   
JHARKHAND          22826                 14186          7896   
KARNATAKA          15056                 16262          7016   
KERALA             20030                  4452           700   
LAKSHADWEEP           20                     2             0   
MADHYA PRADESH     90996                 35608         21090   
MAHARASHTRA        48974                 30368          9696   
MANIPUR             1068                  2606             6   
MEGHALAYA           2642                   670            36   
MIZORAM             2070                    30             8   
NAGALAND             562                   190             2   
ODISHA             30480                 25588         10782   
PUDUCHERRY           208                   306            56   
PUNJAB             14656                 15096          3524   
RAJASTHAN          45684                 66278         11854   
SIKKIM               570                   180             4   
TAMIL NADU         16660                 30908          5060   
TELANGANA           1958                  1422           578   
TRIPURA             5060                  2202           752   
UTTAR PRADESH      51150                135906         57256   
UTTARAKHAND         3752                  6484          1974   
WEST BENGAL        47876                 61158         12308   

                   Assault_on_women_with_intent_to_outrage_her_modesty  \
STATE/UT                                                                 
A & N ISLANDS                                                    600     
ANDHRA PRADESH                                                126952     
ARUNACHAL PRADESH                                               1996     
ASSAM                                                          38256     
BIHAR                                                          16958     
CHANDIGARH                                                      1088     
CHHATTISGARH                                                   47096     
D & N HAVELI                                                     106     
DAMAN & DIU                                                       40     
DELHI                                                          30742     
GOA                                                             1384     
GUJARAT                                                        23186     
HARYANA                                                        17142     
HIMACHAL PRADESH                                                9256     
JAMMU & KASHMIR                                                28638     
JHARKHAND                                                       9898     
KARNATAKA                                                      65750     
KERALA                                                         80438     
LAKSHADWEEP                                                       14     
MADHYA PRADESH                   

## Q6.Find out the top 5 states, where maximum numbers of cases has been reported in TOTAL in span of 2001-2014, each category wise.

### Find Rape

In [67]:
total_case_state.sort_values(by = 'Rape', ascending = False).head(5) #total_case_state is a variable where store total case state wise

Rape  Kidnapping_Abduction  Dowry_Deaths  \
STATE/UT                                                    
MADHYA PRADESH  90996                 35608         21090   
UTTAR PRADESH   51150                135906         57256   
MAHARASHTRA     48974                 30368          9696   
WEST BENGAL     47876                 61158         12308   
RAJASTHAN       45684                 66278         11854   

                Assault_on_women_with_intent_to_outrage_her_modesty  \
STATE/UT                                                              
MADHYA PRADESH                                             195478     
UTTAR PRADESH                                               91212     
MAHARASHTRA                                                114704     
WEST BENGAL                                                 66908     
RAJASTHAN                                                   83362     

                Insult_to_modesty_of_Women  \
STATE/UT                                     
MADHYA PRADESH                       20964   
UTTAR PRADESH                        53130   
MAHARASHTRA                          32634   
WEST BENGAL                           5348   
RAJASTHAN                              794   

                Cruelty_by_Husband_or_his_Relatives  Importation_of_Girls  
STATE/UT                                                                   
MADHYA PRADESH                               102816                   134  
UTTAR PRADESH                                193738                     6  
MAHARASHTRA                                  193202                     6  
WEST BENGAL                                  344124                   254  
RAJASTHAN                                    262200                    14

### Madhya Pradesh has reported the highest number of rape cases in TOTAL in span of 2001-2014

### find Kidnapping_Abduction

In [68]:
total_case_state.sort_values(by='Kidnapping_Abduction',ascending=False).head(5)

Rape  Kidnapping_Abduction  Dowry_Deaths  \
STATE/UT                                                   
UTTAR PRADESH  51150                135906         57256   
RAJASTHAN      45684                 66278         11854   
ASSAM          40190                 62074          3268   
WEST BENGAL    47876                 61158         12308   
BIHAR          30758                 57086         32206   

               Assault_on_women_with_intent_to_outrage_her_modesty  \
STATE/UT                                                             
UTTAR PRADESH                                              91212     
RAJASTHAN                                                  83362     
ASSAM                                                      38256     
WEST BENGAL                                                66908     
BIHAR                                                      16958     

               Insult_to_modesty_of_Women  \
STATE/UT                                    
UTTAR PRADESH                       53130   
RAJASTHAN                             794   
ASSAM                                 254   
WEST BENGAL                          5348   
BIHAR                                 694   

               Cruelty_by_Husband_or_his_Relatives  Importation_of_Girls  
STATE/UT                                                                  
UTTAR PRADESH                               193738                     6  
RAJASTHAN                                   262200                    14  
ASSAM                                       115300                    22  
WEST BENGAL                                 344124                   254  
BIHAR                                        69770                   904

### Uttar Pradesh has reported the highest number of cses under "Kidnapping and Abduction" in TOTAL in span of 2001-2014

### find Dowry_Deaths

In [69]:
total_case_state.sort_values(by='Dowry_Deaths',ascending=False).head(5)

Rape  Kidnapping_Abduction  Dowry_Deaths  \
STATE/UT                                                    
UTTAR PRADESH   51150                135906         57256   
BIHAR           30758                 57086         32206   
MADHYA PRADESH  90996                 35608         21090   
ANDHRA PRADESH  32150                 34504         13844   
WEST BENGAL     47876                 61158         12308   

                Assault_on_women_with_intent_to_outrage_her_modesty  \
STATE/UT                                                              
UTTAR PRADESH                                               91212     
BIHAR                                                       16958     
MADHYA PRADESH                                             195478     
ANDHRA PRADESH                                             126952     
WEST BENGAL                                                 66908     

                Insult_to_modesty_of_Women  \
STATE/UT                                     
UTTAR PRADESH                        53130   
BIHAR                                  694   
MADHYA PRADESH                       20964   
ANDHRA PRADESH                       86964   
WEST BENGAL                           5348   

                Cruelty_by_Husband_or_his_Relatives  Importation_of_Girls  
STATE/UT                                                                   
UTTAR PRADESH                                193738                     6  
BIHAR                                         69770                   904  
MADHYA PRADESH                               102816                   134  
ANDHRA PRADESH                               280906                    34  
WEST BENGAL                                  344124                   254

###  Again Our Analysis Uttar Pradesh has reported the highest number of Deaths caused by Dowry cases in TOTAL in span of 2001-2014, where BIhar,MADHYA PRADESH,ANDHRA PRADESH,WEST BENGAL minimum case with the compare of Uttar Pradesh.


### find Assault_on_women_with_intent_to_outrage_her_modesty

In [71]:
total_case_state.sort_values(by='Assault_on_women_with_intent_to_outrage_her_modesty',ascending=False).head(5)

Rape  Kidnapping_Abduction  Dowry_Deaths  \
STATE/UT                                                    
MADHYA PRADESH  90996                 35608         21090   
ANDHRA PRADESH  32150                 34504         13844   
MAHARASHTRA     48974                 30368          9696   
UTTAR PRADESH   51150                135906         57256   
RAJASTHAN       45684                 66278         11854   

                Assault_on_women_with_intent_to_outrage_her_modesty  \
STATE/UT                                                              
MADHYA PRADESH                                             195478     
ANDHRA PRADESH                                             126952     
MAHARASHTRA                                                114704     
UTTAR PRADESH                                               91212     
RAJASTHAN                                                   83362     

                Insult_to_modesty_of_Women  \
STATE/UT                                     
MADHYA PRADESH                       20964   
ANDHRA PRADESH                       86964   
MAHARASHTRA                          32634   
UTTAR PRADESH                        53130   
RAJASTHAN                              794   

                Cruelty_by_Husband_or_his_Relatives  Importation_of_Girls  
STATE/UT                                                                   
MADHYA PRADESH                               102816                   134  
ANDHRA PRADESH                               280906                    34  
MAHARASHTRA                                  193202                     6  
UTTAR PRADESH                                193738                     6  
RAJASTHAN                                    262200                    14

### MADHYA PRADESH has reported the highest number of 'Assault_on_women_with_intent_to_outrage_her_modesty' in Total in span of 2001-2014 ,where ANDHRA PRADESH,MAHARASHTRA	,UTTAR PRADESH,RAJASTHAN has minimum case compare of MADHYA PRADESH

### find Insult_to_modesty_of_Women

In [72]:
total_case_state.sort_values(by='Insult_to_modesty_of_Women',ascending=False).head(5)

Rape  Kidnapping_Abduction  Dowry_Deaths  \
STATE/UT                                                    
ANDHRA PRADESH  32150                 34504         13844   
UTTAR PRADESH   51150                135906         57256   
MAHARASHTRA     48974                 30368          9696   
TAMIL NADU      16660                 30908          5060   
MADHYA PRADESH  90996                 35608         21090   

                Assault_on_women_with_intent_to_outrage_her_modesty  \
STATE/UT                                                              
ANDHRA PRADESH                                             126952     
UTTAR PRADESH                                               91212     
MAHARASHTRA                                                114704     
TAMIL NADU                                                  43404     
MADHYA PRADESH                                             195478     

                Insult_to_modesty_of_Women  \
STATE/UT                                     
ANDHRA PRADESH                       86964   
UTTAR PRADESH                        53130   
MAHARASHTRA                          32634   
TAMIL NADU                           21170   
MADHYA PRADESH                       20964   

                Cruelty_by_Husband_or_his_Relatives  Importation_of_Girls  
STATE/UT                                                                   
ANDHRA PRADESH                               280906                    34  
UTTAR PRADESH                                193738                     6  
MAHARASHTRA                                  193202                     6  
TAMIL NADU                                    45524                    30  
MADHYA PRADESH                               102816                   134

### ANDHRA PRADESH	has  reported the highest number of Insult_to_modesty_of_Women in Total in span 2001-2014 ,where UTTAR PRADESH,MAHARASHTRA,TAMIL NADU,MADHYA PRADESH has minimum case compare of ANDHRA PRADESH

### Find Cruelty_by_Husband_or_his_Relatives

In [82]:
total_case_state.sort_values(by='Cruelty_by_Husband_or_his_Relatives',ascending=False).head(5)

Rape  Kidnapping_Abduction  Dowry_Deaths  \
STATE/UT                                                    
WEST BENGAL     47876                 61158         12308   
ANDHRA PRADESH  32150                 34504         13844   
RAJASTHAN       45684                 66278         11854   
UTTAR PRADESH   51150                135906         57256   
MAHARASHTRA     48974                 30368          9696   

                Assault_on_women_with_intent_to_outrage_her_modesty  \
STATE/UT                                                              
WEST BENGAL                                                 66908     
ANDHRA PRADESH                                             126952     
RAJASTHAN                                                   83362     
UTTAR PRADESH                                               91212     
MAHARASHTRA                                                114704     

                Insult_to_modesty_of_Women  \
STATE/UT                                     
WEST BENGAL                           5348   
ANDHRA PRADESH                       86964   
RAJASTHAN                              794   
UTTAR PRADESH                        53130   
MAHARASHTRA                          32634   

                Cruelty_by_Husband_or_his_Relatives  Importation_of_Girls  
STATE/UT                                                                   
WEST BENGAL                                  344124                   254  
ANDHRA PRADESH                               280906                    34  
RAJASTHAN                                    262200                    14  
UTTAR PRADESH                                193738                     6  
MAHARASHTRA                                  193202                     6

### WEST BENGAL has reported the highest number of Cruelty_by_Husband_or_his_Relatives in Total in span 2001-2014 ,where ANDHRA PRADESH,RAJASTHAN,UTTAR PRADESH,MAHARASHTRA	 has minimum case compare of WEST BENGAL

### find Importation_of_Girls

In [74]:
total_case_state.sort_values(by='Importation_of_Girls',ascending=False).head(5)

Rape  Kidnapping_Abduction  Dowry_Deaths  \
STATE/UT                                                    
BIHAR           30758                 57086         32206   
JHARKHAND       22826                 14186          7896   
WEST BENGAL     47876                 61158         12308   
MADHYA PRADESH  90996                 35608         21090   
KARNATAKA       15056                 16262          7016   

                Assault_on_women_with_intent_to_outrage_her_modesty  \
STATE/UT                                                              
BIHAR                                                       16958     
JHARKHAND                                                    9898     
WEST BENGAL                                                 66908     
MADHYA PRADESH                                             195478     
KARNATAKA                                                   65750     

                Insult_to_modesty_of_Women  \
STATE/UT                                     
BIHAR                                  694   
JHARKHAND                              566   
WEST BENGAL                           5348   
MADHYA PRADESH                       20964   
KARNATAKA                             2276   

                Cruelty_by_Husband_or_his_Relatives  Importation_of_Girls  
STATE/UT                                                                   
BIHAR                                         69770                   904  
JHARKHAND                                     23910                   298  
WEST BENGAL                                  344124                   254  
MADHYA PRADESH                               102816                   134  
KARNATAKA                                     72706                    94

### BIHAR  has reported the highest number of Importation_of_Girls in Total in span 2001-2014 ,where JHARKHAND	, WEST BENGAL, MADHYA PRADESH	,KARNATAKA has minimum case compare of BIHAR

## Q7.Which state has featured in both the lists of "Maximum number of rape cases" and "Maximum number of Importation cases".

In [78]:
max_rape_importaion_grl=highest_importation.merge(highest_rape)

In [79]:
max_rape_importaion_grl

STATE/UT  Year  Rape  Kidnapping_Abduction  Dowry_Deaths  \
0  MADHYA PRADESH  2011  3406                  1088           811   

   Assault_on_women_with_intent_to_outrage_her_modesty  \
0                                               6665     

   Insult_to_modesty_of_Women  Cruelty_by_Husband_or_his_Relatives  \
0                         762                                 3732   

   Importation_of_Girls  
0                    45

### According to our Analysis MADHYA PRADESH has  Maximum number  of rape  and  Importation of girls cases.

## Q8.Which state has featured in both the lists of "Maximum number of rape cases" and "Maximum number of Deaths due to Dowry cases".

In [80]:
max_rape_and_Death=highest_death.merge(highest_rape)

In [81]:
max_rape_and_Death

STATE/UT  Year  Rape  Kidnapping_Abduction  Dowry_Deaths  \
0  UTTAR PRADESH  2014  3467                 10626          2469   

   Assault_on_women_with_intent_to_outrage_her_modesty  \
0                                               8605     

   Insult_to_modesty_of_Women  Cruelty_by_Husband_or_his_Relatives  \
0                          29                                10471   

   Importation_of_Girls  
0                     0

### According to our AnalysisUTTAR PRADESH has Maximum number of rape and Dowry_Deaths cases.